modules/data.atmosphere/R/download_caladapt_loca.R

```diff
@@ -0,0 +1,297 @@
+#' Retrieve Cal-Adapt climate raster for a polygon or a point
+#'
+#' @description
+#' Fetches climate raster data from Cal-Adapt for a specified polygon or point,
+#' GCM, emission scenario, and time period. Supports LOCA (CMIP5-based) data,
+#' and ensures that the geometry is within the bounding box of valid coordinates
+#' for California.
+#'
+#' @references
+#' - [Cal-Adapt](https://caladapt.org/)
+#' - [Cal Adapt Data Catalog Documentation](https://berkeley-gif.github.io/caladapt-docs/data-catalog.html)
+#' - [Cal-Adapt Loca Data Catalog](https://albers.cnr.berkeley.edu/data/scripps/loca/met/)
+#'
+#' @param sf_obj An `sf`, `sfc`, or `SpatVector` row representing the points or polygon boundary
+#'   (required for `download_caladapt_loca_raster()`).
+#' @param var Character. Climate variable to retrieve. One of "tasmax", "tasmin", "pr",
+#' "swe", "baseflow", "et", "rainfall",
+#' "runoff", "snowfall", "soilMoist1", "Tair".
+#' @param gcm Character. GCM name. One of "HadGEM2-ES", "CNRM-CM5", "CanESM2", "MIROC5", "ACCESS1-0",
+#' "CCSM4", "CESM1-BGC", "CMCC-CMS", "GFDL-CM3", "HadGEM2-CC", "ens32avg",
+#' "ens32max", "ens32min".
+#' @param scenario Character. Emission scenario. One of "historical", "rcp45", "rcp85". Historical (default)
+#' period covers 1950-2005. RCP45 and RCP85 scenarios cover 2006-2100.
+#' @param period Character. Time aggregation. One of "day", "month", "year", "30yavg".
+#' @param start_year Numeric. Start year for data retrieval.
+#' @param end_year Numeric. End year for data retrieval.
+#' @param out_dir Character. Output directory path to store downloaded data.
+#' @param raster_path Character. (Optional) File path to a local raster file. If provided, the function
+#' will load the raster from this path instead of using the Cal-Adapt API.
+#'
+#' @return A tibble row with appended climate raster in the `raster` column.
+#' For points, the row will also include `lat` and `lon` columns.
+#'
+#' @examples
+#' \dontrun{
+#' # Polygon-level example
+#' yolo_polygon <- ca_aoipreset_geom("counties") |>
+#'   filter(state_name == "California") |>
+#'   filter(name == "Yolo") |>
+#'   select(state_name, county_name = name, geom) |>
+#'   st_transform(4326)
+#'
+#' polygon_rasters <- download_caladapt_loca(
+#'   sf_obj = yolo_polygon,
+#'   var = "pr",
+#'   gcm = "CNRM-CM5",
+#'   scenario = "rcp85",
+#'   period = "year",
+#'   start_year = 2006, end_year = 2100,
+#'   out_dir = "data"
+#' )
+#' coords <- data.frame(lat = 36, lon = -120) |>
+#'   st_as_sf(coords = c("lon", "lat"), crs = 4326)
```

Comment:
should this be `sf::st_as_sf`, or maybe should there be a library call in the example?